components/opentherm/output/output.cpp

```diff
@@ -1,11 +1,12 @@
+#include <algorithm> // Some compilers may need this for std::clamp
 #include "output.h"
 
 namespace esphome {
 namespace opentherm {
 
 void opentherm::OpenthermOutput::write_state(float state) {
   ESP_LOGD("opentherm.output", "Received state: %.2f. Min value: %.2f, max value: %.2f", state, min_value_, max_value_);
-  this->state = state < 0.003 && this->zero_means_zero_ ? 0.0 : min_value_ + state * (max_value_ - min_value_);
+  this->state = state < 0.003 && this->zero_means_zero_ ? 0.0 : std::clamp(min_value_ + state * (max_value_ - min_value_), min_value_, max_value_);
```

Comment:
Maybe use `clamp` from `esphome/core/helpers.h`? It uses STL implementation when it's available. I'm not sure what are the cases when it's not available, but I guess that helper is there for a reason :)

Reply:
Hadn't realised there were "helpers" reimplementing functions that _might_ be missing.